src/service/resource.js

```diff
@@ -159,6 +159,9 @@ export class Resource {
     /** @private {boolean} */
     this.isFixed_ = false;
 
+    /** @private {boolean} */
+    this.isInFirstViewport_ = false;
```

Comment:
I think this value is too special. Should we just calculate it where it's needed based on the available `iniLayoutBox`?